modin/core/dataframe/pandas/dataframe/protocol/column.py

```diff
@@ -0,0 +1,590 @@
+                                categorical values to other objects exists
+            - "mapping" : dict, Python-level only (e.g. ``{int: str}``).
+                          None if not a dictionary-style categorical.
+
+        Raises
+        ------
+        ``RuntimeError`` if the dtype is not categorical.
+        """
+        if not self.dtype[0] == DTypeKind.CATEGORICAL:
+            raise TypeError(
+                "`describe_categorical only works on a column with "
+                "categorical dtype!"
+            )
+
+        # TODO: Raise an exception if ``self._allow_copy==False``?
+        pandas_series = self._col.to_pandas().squeeze(axis=1)
+        ordered = pandas_series.dtype.ordered
+        is_dictionary = True
+        # NOTE: this shows the children approach is better, transforming
+        # `categories` to a "mapping" dict is inefficient
+        # codes = self._col.values.codes  # ndarray, length `self.size`
+        # categories.values is ndarray of length n_categories
+        # TODO: Raise an exception if ``self._allow_copy==False``?
+        categories = pandas_series.values.categories.values
+        mapping = {ix: val for ix, val in enumerate(categories)}
+        return ordered, is_dictionary, mapping
+
+    @property
+    def describe_null(self) -> Tuple[int, Any]:
+        """
+        Return the missing value (or "null") representation the column dtype uses.
+
+        Return as a tuple ``(kind, value)``.
+
+        * Kind:
+            - 0 : non-nullable
+            - 1 : NaN/NaT
+            - 2 : sentinel value
+            - 3 : bit mask
+            - 4 : byte mask
+        * Value : if kind is "sentinel value", the actual value. If kind is a bit
+          mask or a byte mask, the value (0 or 1) indicating a missing value. None
+          otherwise.
+
+        Returns
+        -------
+        tuple
+            ``(kind, value)``.
+        """
+        _k = DTypeKind
+        kind = self.dtype[0]
+        value = None
+        if kind == _k.FLOAT:
+            null = 1  # np.nan
+        elif kind == _k.DATETIME:
+            null = 1  # np.datetime64('NaT')
+        elif kind in (_k.INT, _k.UINT, _k.BOOL):
+            # TODO: check if extension dtypes are used once support for them is
+            #       implemented in this protocol code
+            null = 0  # integer and boolean dtypes are non-nullable
+        elif kind == _k.CATEGORICAL:
+            # Null values for categoricals are stored as `-1` sentinel values
+            # in the category date (e.g., `col.values.codes` is int8 np.ndarray)
+            null = 2
+            value = -1
+        elif kind == _k.STRING:
+            null = 4
+            value = (
+                0  # follow Arrow in using 1 as valid value and 0 for missing/null value
+            )
+        else:
+            raise NotImplementedError(f"Data type {kind} not yet supported")
+
+        return null, value
+
+    @property
+    def null_count(self) -> int:
```

Comment:
can we make this property to be caching its result?

Reply:
Yes, we can, fixed